packages/api/src/hooks/p2p/useAdvertiserRelationsAddBlocked.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback } from 'react';
+import useInvalidateQuery from '../../useInvalidateQuery';
+import useMutation from '../../useMutation';
+
+/** This hook blocks advertisers of the current user by passing the advertiser id. */
+const useAdvertiserRelationsAddBlocked = () => {
+    const invalidate = useInvalidateQuery();
+    const { data, mutate, ...rest } = useMutation('p2p_advertiser_relations', {
+        onSuccess: () => invalidate('p2p_advertiser_relations'),
```

Comment:
I dont feel this is right, we are invalidation `p2p_advertiser_relations` multiple times here. Imagine it in the component, this will be called so many times. I suggest to have this `useMutation` in `useAdvertiserRelations` and we can extend the data and mutate from there to here. In this way we only invalidate once.

Reply:
Thanks for the suggestion will change that 🙏